src/components/Table/Table.test.tsx

```diff
@@ -199,4 +199,85 @@ describe('<Table />', () => {
       });
     });
   });
+
+  describe('onDataClick', () => {
+    it('should trigger onDataClick handler when clicking data', () => {
+      const onDataClickCallback = jest.fn();
+      const onDataClickMobileCallback = jest.fn();
+      const wrapper = enzyme.mount(
+        <Table data={[{ a: 1, b: 3, c: 5 }, { a: 2, b: 4, c: 8 }]} onDataClick={onDataClickCallback} />,
+      );
+      const wrapperMobileView = enzyme.mount(
+        <Table
+          mode="card"
+          data={[{ a: 1, b: 3, c: 5 }, { a: 2, b: 4, c: 8 }]}
+          onDataClick={onDataClickMobileCallback}
+        />,
+      );
+
+      wrapper.find('td').forEach(td => td.simulate('click'));
+      expect(onDataClickCallback).toHaveBeenCalledTimes(6);
+
+      wrapperMobileView
+        .find('div')
+        .filterWhere(item => typeof item.prop('onClick') === 'function')
+        .forEach(item => item.simulate('click'));
```

Comment:
There is a better to way to check the functionality in a mobile view. You can dynamically change the prop of the mounted  component `wrapper.setProps({ mode: 'card' });`. Doing this allows you to check `toHaveBeenCalledTimes(12)` only once at the last line.

Reply:
same below